torchvision/csrc/io/decoder/decoder.cpp

```diff
@@ -505,10 +484,14 @@ int Decoder::getFrame(size_t workingTimeInMs) {
   // once decode() method gets called and grab some bytes
   // run this method again
   // init package
-  AVPacket avPacket;
-  av_init_packet(&avPacket);
-  avPacket.data = nullptr;
-  avPacket.size = 0;
+  // update 03/22: moving memory management to ffmpeg
+  AVPacket* avPacket;
+  avPacket = av_packet_alloc();
```

Comment:
according to LeakSanitizer this call will result in a memory leak.

Reply:
We should call `av_packet_free(&avPacket);` when we are done with it. We need to be careful and add this to all the places where we return early too.

Reply:
Yeah, I was thinking about this, but my thought was that `av_packet_unref` would have freed whatever was allocated (which seems to have been an incorrect assumption). 

I'll fix this ASAP